vpd-tool/src/vpd_tool.cpp

```diff
@@ -1,20 +1,73 @@
 #include "vpd_tool.hpp"
 
+#include "tool_constants.hpp"
+#include "tool_types.hpp"
+#include "utils.hpp"
+
+#include <iostream>
+
 namespace vpd
 {
-int VpdTool::readKeyword(const std::string& i_fruPath,
+int VpdTool::readKeyword(const std::string& i_vpdPath,
                          const std::string& i_recordName,
                          const std::string& i_keywordName,
                          const bool i_onHardware,
                          const std::string& i_fileToSave)
 {
-    // ToDo: Need to add implementation details
-    (void)i_fruPath;
-    (void)i_recordName;
-    (void)i_keywordName;
-    (void)i_onHardware;
-    (void)i_fileToSave;
-
-    return 0;
+    int l_rc = -1;
+    try
+    {
+        types::DbusVariantType l_keywordValue;
+        if (i_onHardware)
+        {
+            // TODO: Implement read keyword's value from hardware
+        }
+        else
+        {
+            std::string l_inventoryObjectPath(constants::baseInventoryPath +
+                                              i_vpdPath);
+
+            l_keywordValue = utils::readDbusProperty(
+                constants::inventoryManagerService, l_inventoryObjectPath,
+                constants::ipzVpdInfPrefix + i_recordName, i_keywordName);
+        }
+
+        if (const auto l_value =
```

Comment:
l_keywordValue can be empty as well, which shows that there was some error in Dbus operation? Should be try to identify that?

Reply:
updated.